### PR TITLE
Infer Large Integers as Float64 in arrow-csv

### PR DESCRIPTION
**Pull request by isnlan**
This modification can prevent the incorrect parsing of overly long numeric string types, avoiding errors when interpreting them as int64.

### CODE REVIEW

**Review by viirya — COMMENTED** [review]
Could you add a test for this? Thanks.

File: arrow-csv/src/reader/mod.rs [hunk 1]
```diff
@@ -147,7 +147,7 @@ lazy_static! {
     /// Order should match [`InferredDataType`]
     static ref REGEX_SET: RegexSet = RegexSet::new([
         r"(?i)^(true)$|^(false)$(?-i)", //BOOLEAN
-        r"^-?(\d+)$", //INTEGER
+        r"^-?\d{1,19}$", //INTEGER
```
Comment (tustvold) [inline arrow-csv/src/reader/mod.rs:150]: If I'm not mistaken this changes the schema inference to infer a float for such types, I'm not sure about this?

**Comment by tustvold** [comment]
Changing to draft to indicate this is not currently waiting for review

**Comment by tustvold** [comment]
Inferring that column as Float64 will result in silent precision loss though? Is this not a worse behaviour?

**Comment by kskalski** [comment]
It's debatable whether this behaviour is worse or still an improvement:
* at least it enables data to be processed and some use-cases will be fine with lost precision
* it would be consistent with `json` inference logic as it is now
* agree totally though that silent precision loss is a big red flag

An ultimate solution would be to support `decimal128` / `decimal256` (I recently filed a similar request for json in https://github.com/apache/arrow-rs/issues/4856). All in all, I guess I'm still a fan of this PR if that would be faster to get in.

**Comment by tustvold** [comment]
Closing this as I believe the current behaviour is consistent with other arrow implementations

**Pull request closed without merging**